### PR TITLE
Added support for sqlcipher to be used with sqlite

### PR DESCRIPTION
**Pull request by oshoemaker**
This commit adds support for sqlcipher for database encrypition when using sqlite

### CODE REVIEW

**Comment by oshoemaker** [comment]
Ok, after closing and opening all checks passed.  It seems that travis-ci is having some load issues and is failing on DB connection timeouts periodically.  Hopefully this PR is ready to go.

File: lib/dialects/sqlite/connection-manager.js [hunk 1]
```diff
@@ -55,6 +55,15 @@ ConnectionManager.prototype.getConnection = function(options) {
       // explicitly disallowed. It's still opt-in per relation
       connection.run('PRAGMA FOREIGN_KEYS=ON');
     }
+
+    // PRAGMA commands to support sqlcipher database encryption
+    if (self.sequelize.options.key) {
+      connection.run('PRAGMA KEY = \'' + self.sequelize.options.key + '\'');
+    }
```
Comment (mickhansen) [inline lib/dialects/sqlite/connection-manager.js]: Looking at it again, `key` might be a tad generic?

Comment (oshoemaker) [inline lib/dialects/sqlite/connection-manager.js]: Would you like a object in the opts of:

sqlcipher: {
  key: '<key>',
  cipher: '<cipher'
}

Comment (mickhansen) [inline lib/dialects/sqlite/connection-manager.js]: yeah sounds good.

**Comment by janmeier** [comment]
Ping @oshoemaker

**Comment by oshoemaker** [comment]
Updated to use an object in the config for sqlcipher.

File: lib/dialects/sqlite/connection-manager.js [hunk 2]
```diff
@@ -55,6 +55,18 @@ ConnectionManager.prototype.getConnection = function(options) {
       // explicitly disallowed. It's still opt-in per relation
       connection.run('PRAGMA FOREIGN_KEYS=ON');
     }
+
+    if (self.sequelize.options.sqlcipher) {
+      // PRAGMA commands to support sqlcipher database encryption
+      if (self.sequelize.options.key) {
+        connection.run('PRAGMA KEY = \'' + self.sequelize.options.sqlcipher.key + '\'');
+      }
```
Comment (mickhansen) [inline lib/dialects/sqlite/connection-manager.js]: The if checks are not correct (use the old format).

**Comment by oshoemaker** [comment]
How about this one.

1c7ac28

**Comment by mickhansen** [comment]
Keep the `if self.sequelize.options.sqlcipher` around, cleaner with no repeats.

**Comment by McPo** [comment]
Any update on getting this PR merged. Is such a feature desirable to the maintainers?

**Comment by janmeier** [comment]
Closed in favor of https://github.com/sequelize/sequelize/pull/6020/files

**Pull request closed without merging**